app/controllers/Application.java

```diff
@@ -1869,6 +1909,88 @@ public static Result getJobBestParameter(String jobDefId) {
     bestParamValueMap.put("executionId", bestTuningJobExecutionParamSet.jobExecution.jobExecUrl);
     return ok(Json.toJson(bestParamValueMap));
   }
+
+  /** Rest Api implementation for authenticating the user using Scheduler's Authentication API
+   * @result JSON containing session.id on successful authentication else the error message
+   * eg: Formatted JSON Data
+   * {
+   *  "status" : "success"
+   *  "session_id" : "4b52bfe1-5a22-9a56-8ced-955dd5b9f0ea"
+   *}
+   **/
+  public static Result authenticateUser() {
+    DynamicForm form = Form.form().bindFromRequest(request());
+    final String username = form.get(USERNAME);
+    final String password = form.get(PASSWORD);
+    final String schedulerUrl = form.get(SCHEDULER_URL);
+    logger.info("Authenticating user: " + username + " " + schedulerUrl);
+    logger.info(isSet(username) + " " + isSet(password) + " " + isSet(schedulerUrl));
+    if (!isSet(username) || !isSet(password)) {
+      return badRequest("Username or Password cannot be empty");
+    } else if (!isSet(schedulerUrl)) {
+      return badRequest("Scheduler Url is empty!!");
+    }
+    Map<String, String> response;
+    try {
+      response = authenticateUser(username, password, schedulerUrl + AZKABAN_AUTHENTICATION_URL_SUFFIX);
+    } catch (Exception ex) {
+      logger.error("Some error occured while authenticating the user " + ex);
```

Comment:
Better to pass exception as second argument to logger.error

Reply:
Sure, missed it.